src/foam/dao/CSVSink.js

```diff
@@ -85,9 +85,11 @@ foam.CLASS({
     {
       name: 'eof',
       code: function() {
+        if ( this.outputter.isFirstRow ) this.outputter.outputHeader(this.__context__);
```

Comment:
Rather than doing it here, what if you did it in the toString() method of CSVOutputterImpl? Then the interface doesn't need to change and it can deal with the business logic of checking `isFirstRow`

Reply:
🥇 🎉